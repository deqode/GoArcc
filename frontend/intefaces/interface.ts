import { NextApiRequest } from 'next'

export interface UserResponse {
  accessToken: string
  idToken: string
  userId: string
}

export interface ResponseError<T> {
  error: boolean
  message: string
  data?: T | null
}

export interface UserAccount {
  id: string
  slug: string
  userId: string
}

export interface IronSessionRequest extends NextApiRequest {
  session: {
    //TODO:remove any
    get: (arg0: string) => any
    set: (arg0: string) => any
  }
}
