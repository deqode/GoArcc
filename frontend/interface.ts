export interface UserContextInterface {
  user: {
    accessToken: string
    idToken: string
    userId: string
    provider: string
    accounts: Array<any>
    state: number
    //  loading=0
    //  authenticated=1
    //  unauthenticated=-1
  }
  setUser(value: any): void
  removeUser(): void
}

export interface AppContextInterface {
  app: {
    centrifuge: any
    subscribed: boolean
  }
  setApp(value: any): void
  removeApp(): void
}
