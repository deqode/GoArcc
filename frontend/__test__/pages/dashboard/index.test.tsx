import renderer from 'react-test-renderer'

import { UserResponse } from '../../../interface'
import Dashboard, { handler } from '../../../pages/dashboard'

describe('dashboard/index page tests', () => {
  const mockUser: UserResponse = {
    accessToken: 'mockAccessToken',
    idToken: 'mockidToken',
    userId: 'mockuserId',
  }
  it('should match the snapshot', () => {
    const tree = renderer.create(<Dashboard user={mockUser} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should return to landing page if session not available', async () => {
    const req = {
      session: { get: jest.fn(() => undefined), save: jest.fn() },
    }
    const response = await handler({ req })
    expect(response).toStrictEqual({
      redirect: {
        permanent: false,
        destination: '/',
      },
    })
  })
  it('should return user if session is available', async () => {
    const req = {
      session: { get: jest.fn(() => 'MockUser'), save: jest.fn() },
    }
    const response = await handler({ req })
    expect(response).toStrictEqual({ props: { user: 'MockUser' } })
  })
})
