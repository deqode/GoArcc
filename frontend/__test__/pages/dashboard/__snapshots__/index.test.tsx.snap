// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dashboard/index page tests should match the snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation0 MuiPaper-rounded"
>
  <div
    className="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-around"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-4 MuiGrid-grid-xl-4"
    >
      <div
        className="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation0 MuiPaper-rounded"
      >
        <h1
          className="MuiTypography-root MuiTypography-h3 MuiTypography-colorPrimary"
        >
          dashboard-index:GREETING
        </h1>
        <br />
        <h5
          className="MuiTypography-root MuiTypography-h5 MuiTypography-colorSecondary"
        >
          dashboard-index:CONNECT_GITHUB
        </h5>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-4 MuiGrid-grid-xl-4"
    >
      <div
        className="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation0 MuiPaper-rounded"
      >
        <p
          className="MuiTypography-root makeStyles-text-3 MuiTypography-body1 MuiTypography-displayInline"
        >
          <div
            className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
            role="progressbar"
            style={
              Object {
                "height": 40,
                "width": 40,
              }
            }
          >
            <svg
              className="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                cx={44}
                cy={44}
                fill="none"
                r={20.2}
                strokeWidth={3.6}
                style={Object {}}
              />
            </svg>
          </div>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`dashboard/index page tests should match the snapshot with token error 1`] = `null`;

exports[`dashboard/index page tests should match the snapshot with user 1`] = `null`;

exports[`dashboard/index page tests should match the snapshot with user vcs network error 1`] = `null`;
